src/cmd/restart.go

```diff
@@ -6,49 +6,65 @@ import (
 	"os/exec"
 	"path/filepath"
 	"runtime"
+	"strings"
 )
 
-// EvalSpotifyRestart Restarts/starts spotify
-func EvalSpotifyRestart(start bool, flags ...string) {
-	launchFlag := settingSection.Key("spotify_launch_flags").Strings("|")
-	if len(launchFlag) > 0 {
-		flags = append(flags, launchFlag...)
-	}
-
-	enableDevtools := settingSection.Key("always_enable_devtools").MustBool(false)
-	if enableDevtools {
-		SetDevTools()
-	}
-
+func SpotifyKill() {
 	switch runtime.GOOS {
 	case "windows":
 		isRunning := exec.Command("tasklist", "/FI", "ImageName eq spotify.exe")
 		result, _ := isRunning.Output()
-		if !bytes.Contains(result, []byte("No tasks are running")) || start {
+		if !bytes.Contains(result, []byte("No tasks are running")) {
 			exec.Command("taskkill", "/F", "/IM", "spotify.exe").Run()
-			if isAppX {
-				ps, _ := exec.LookPath("powershell.exe")
-				exe := filepath.Join(os.Getenv("LOCALAPPDATA"), "Microsoft", "WindowsApps", "Spotify.exe")
-				flags = append([]string{"-NoProfile", "-NonInteractive", `& "` + exe + `" --app-directory="` + appDestPath + `"`}, flags...)
-				exec.Command(ps, flags...).Start()
-			} else {
-				exec.Command(filepath.Join(spotifyPath, "spotify.exe"), flags...).Start()
-			}
 		}
 	case "linux":
 		isRunning := exec.Command("pgrep", "spotify")
 		_, err := isRunning.Output()
-		if err == nil || start {
+		if err == nil {
 			exec.Command("pkill", "spotify").Run()
-			exec.Command(filepath.Join(spotifyPath, "spotify"), flags...).Start()
 		}
 	case "darwin":
 		isRunning := exec.Command("sh", "-c", "ps aux | grep 'Spotify' | grep -v grep")
 		_, err := isRunning.CombinedOutput()
-		if err == nil || start {
+		if err == nil {
 			exec.Command("pkill", "Spotify").Run()
-			flags = append([]string{"-a", "/Applications/Spotify.app", "--args"}, flags...)
-			exec.Command("open", flags...).Start()
 		}
 	}
 }
+
+func SpotifyStart(flags ...string) {
+	launchFlag := settingSection.Key("spotify_launch_flags").Strings("|")
+	if len(launchFlag) > 0 {
+		flags = append(flags, launchFlag...)
+	}
+
+	switch runtime.GOOS {
+	case "windows":
+		if isAppX {
+			ps, _ := exec.LookPath("powershell.exe")
+			exe := filepath.Join(os.Getenv("LOCALAPPDATA"), "Microsoft", "WindowsApps", "Spotify.exe")
+			cmd := `& "` + exe + `" --app-directory="` + appDestPath + `"`
+			if len(flags) > 0 {
+				cmd += " " + strings.Join(flags, " ")
+			}
+			exec.Command(ps, "-NoProfile", "-NonInteractive", "-Command", cmd).Start()
+		} else {
+			exec.Command(filepath.Join(spotifyPath, "spotify.exe"), flags...).Start()
+		}
+	case "linux":
+		exec.Command(filepath.Join(spotifyPath, "spotify"), flags...).Start()
+	case "darwin":
+		flags = append([]string{"-a", "/Applications/Spotify.app", "--args"}, flags...)
+		exec.Command("open", flags...).Start()
+	}
+}
+
+func SpotifyRestart(flags ...string) {
+	enableDevtools := settingSection.Key("always_enable_devtools").MustBool(false)
```

Comment:
This should also work in `SpotifyStart`... Some commands start spotify for a reason